facebook/app/controllers/comments_controller.rb

```diff
@@ -1,8 +1,8 @@
 class CommentsController < ApplicationController
   def create
     @post = Post.find(params[:post_id])
-    @comment = @post.comments.create(comment_params)
-    @comment.update(user: current_user)
+    @comment = @post.comments.build(comment_params)
+    @comment.user = current_user
     if @comment.save
```

Comment:
even when we have an else?

Reply:
Usually, yes, because many different exceptions can happen when using `.save!` or `.create!` (mostly model validations) and we want to capture and handle them differently in a real project. An `else` is not enough to handle that on large projects.

Reply:
ok so that's a refactor I'll work through as I update my previous projects and get them into a repo

Reply:
updated in 4 places